src/ui/ParticleEngine.js

```diff
@@ -215,8 +216,10 @@ exports = Class(View, function () {
    * returns an array populated with n particle objects
    * modify each particle object, then pass the array in via this.emitParticles
    */
-  this.obtainParticleArray = function (count) {
-    for (var i = 0; i < count; i++) {
+  this.obtainParticleArray = function (count, performance) {
```

Comment:
The parameter `performance` should be `opts` in case we want to further extend it. In general, we follow a pattern for parameters where the required parameters are listed with their own slot, i.e. `count` in this case, and any optional params are delivers as properties on an optional `opts` object